src/python/gudhi/barycenter.py

```diff
@@ -0,0 +1,342 @@
+    themin = min(xmin, ymin)
+    themax = max(xmax, ymax)
+    ax.set_xlim(themin, themax)
+    ax.set_ylim(themin, themax)
+    ax.add_patch(Polygon([[themin,themin], [themax,themin], [themax,themax]], fill=True, color='lightgrey'))
+    ax.set_xticks([])
+    ax.set_yticks([])
+    ax.set_aspect('equal', adjustable='box')
+    ax.set_title("Estimated barycenter")
+
+    plt.show()
+
+
+def _test_perf():
+    nb_repeat = 10
+    nb_points_in_dgm = [5, 10, 20, 50, 100]
+    nb_dmg = [3, 5, 10, 20]
+
+    from time import time
+    for m in nb_dmg:
+        for n in nb_points_in_dgm:
+            tstart = time()
+            for _ in range(nb_repeat):
+                X = [np.random.rand(n, 2) for _ in range(m)]
+                for diag in X:
+                    #enforce having diagrams
+                    diag[:,1] = diag[:,1] + diag[:,0]
+                _ = lagrangian_barycenter(X)
+            tend = time()
+            print("Computation of barycenter in %s sec, with k = %s diags and n = %s points per diag."%(np.round((tend - tstart)/nb_repeat, 2), m, n))
+            print("********************")
+
+
+def _sanity_check(verbose):
+    #dg1 = np.array([[0.2, 0.5]])
+    #dg2 = np.array([[0.2, 0.7], [0.73, 0.88]])
+    #dg3 = np.array([[0.3, 0.6], [0.7, 0.85], [0.2, 0.3]])
+    #X = [dg1, dg2, dg3]
+    #Y, a = lagrangian_barycenter(X, verbose=verbose)
+    #_plot_barycenter(X, Y, a)
+
+    #dg1 = np.array([[0.2, 0.5]])
+    #dg2 = np.array([]) # The empty diagram
+    #dg3 = np.array([[0.4, 0.8]])
+    #X = [dg1, dg2, dg3]
+    #Y, a = lagrangian_barycenter(X, verbose=verbose)
+    #_plot_barycenter(X, Y, a)
+ 
+    #dg1 = np.array([])
+    #dg2 = np.array([]) # The empty diagram
+    #dg3 = np.array([])
+    #X = [dg1, dg2, dg3]
+    #Y, a = lagrangian_barycenter(X, verbose=verbose)
+    #_plot_barycenter(X, Y, a)
+    #print(Y)
+ 
+    dg1 = np.array([[0.1, 0.12], [0.21, 0.7], [0.4, 0.5], [0.3, 0.4], [0.35, 0.7], [0.5, 0.55], [0.32, 0.42], [0.1, 0.4], [0.2, 0.4]])
+    dg2 = np.array([[0.09, 0.11], [0.3, 0.43], [0.5, 0.61], [0.3, 0.7], [0.42, 0.5], [0.35, 0.41], [0.74, 0.9], [0.5, 0.95], [0.35, 0.45], [0.13, 0.48], [0.32, 0.45]])
+    dg3 = np.array([[0.1, 0.15], [0.1, 0.7], [0.2, 0.22], [0.55, 0.84], [0.11, 0.91], [0.61, 0.75], [0.33, 0.46], [0.12, 0.41], [0.32, 0.48]])
+    dg4 = np.array([])
+    X = [dg4]
+    Y, a = lagrangian_barycenter(X, verbose=verbose)
+    #_plot_barycenter(X, Y, a)
+    print(Y)
+    print(np.array_equal(Y, np.empty(shape=(0,2) )))
+ 
+    
+    #dg1 = np.array([[0.2, 0.5]])
+    #dg2 = np.array([[0.2, 0.7]])
+    #dg3 = np.array([[0.3, 0.6], [0.7, 0.8], [0.2, 0.3]])
+    #dg4 = np.array([])
+    #
+    #bary, a = lagrangian_barycenter(pdiagset=[dg1, dg2, dg3, dg4],init=3, verbose=True)
+    #_plot_barycenter([dg1, dg2, dg3, dg4], bary, a)
+    #message = "Wasserstein barycenter estimated:"    
+    #print(message)
+    #print(bary)
+
+if __name__=="__main__":
+    _sanity_check(verbose = True)
```

Comment:
Ah, so you have a lot of testing code (and a lot is commented out) in the source file itself, and running the file runs the tests. It seems to be a somewhat common practice in python, but since we haven't used it in gudhi yet, I'd like to understand if this is useful.
The part of _sanity_check that currently doesn't have a `#` looks like it might as well go in the separate test file. Is this function meant to be run regularly as part of the testsuite, or only when debugging manually? What about _plot_barycenter, is it meant for debugging? For an example or tutorial? As a public interface?

Reply:
Hum... I am not sure what follows "__test__": is useful actually. It is mostly an artifact coming from my trials. If this is not a spread practice in gudhi, I should remove it.

Regarding _plot_barycenter, I think this function should be public at some point as it could be convenient, but I didn't know which approach I should take. Currently, it is somewhat cumbersome to use, so I preferred to hide it.

Reply:
If you consider some code useless, then yes, please remove it.
For the plot function, moving it to an example could be a reasonable compromise.

Reply:
What do you mean by "moving it to an example"? In the docs?

Reply:
There is a directory `example/`, next to `test/`, where you can add files. Or you can write a tutorial for https://github.com/GUDHI/TDA-tutorial .

Reply:
@mglisse : I will write a tutorial, it seems to be a good idea as there are some "maths" involved. I was wondering, in a tutorial, should I consider the user have access to the "live" version of gudhi? (i.e. with `lagrangian_barycenter` in this case). I am afraid that it could be somewhat misleading.